crates/lib/src/qpu/readout_data.rs

```diff
@@ -0,0 +1,83 @@
+//! This modules provides types and functions for initializing and working with
+//! QPU readout data.
+use enum_as_inner::EnumAsInner;
+use num::complex::Complex64;
+use quil_rs::instruction::MemoryReference;
+use std::collections::HashMap;
+
+use qcs_api_client_grpc::models::controller::{
+    readout_values as controller_readout_values, ReadoutValues as ControllerReadoutValues,
+};
+
+/// A row of readout values from the QPU.
+#[derive(Debug, Clone, EnumAsInner, PartialEq)]
+pub enum ReadoutValues {
+    /// Integer readout values
+    Integer(Vec<i64>),
+    /// Real numbered readout values
+    Real(Vec<f64>),
+    /// Complex readout values
+    Complex(Vec<Complex64>),
+}
+
+/// This struct encapsulates the readout data returned from the QPU after executing a job.
+#[derive(Debug, Clone, PartialEq)]
+pub struct QpuReadout {
```

Comment:
When we start to handle memory values alongside readout, both coming back from QPU execution - which we could start to do today, and will do more in the future - would those live on this struct, or another one? If they'd be here, which makes sense to me, then this should have a name that expresses it's the resulting data of QPU execution and not just readout.

Reply:
Renamed this `QpuResultData` and updated the module to match.